tensorflow/contrib/learn/python/learn/datasets/mnist.py

```diff
@@ -223,23 +223,25 @@ def fake():
   TEST_IMAGES = 't10k-images-idx3-ubyte.gz'
   TEST_LABELS = 't10k-labels-idx1-ubyte.gz'
 
-  local_file = base.maybe_download(TRAIN_IMAGES, train_dir,
-                                   SOURCE_URL + TRAIN_IMAGES)
+  def download_local_file(fname):
+    try:
+      return base.maybe_download(fname, train_dir, SOURCE_URL + fname)
```

Comment:
Cool, that sounds like a great way of resolving the issue. I am a bit busy for the rest of today but I will try to take a crack at it in the next day or so (unless someone else wants to jump in).